portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/security/login/PasswordEncryptor.testcase

```diff
@@ -0,0 +1,65 @@
+@component-name = "portal-security"
+definition {
+
+	property app.server.types = "jboss,tcserver,tomcat,weblogic,websphere,wildfly";
+	property database.types = "db2,hypersonic,mariadb,mysql,oracle,postgresql,sqlserver,sybase";
+	property portal.upstream = "gabor";
+	property testray.main.component.name = "Login";
+
+	setUp {
+		task ("Setup: Setup portal instance") {
+			TestCase.setUpPortalInstance();
+		}
+	}
+
+	@description = "Users could login through the UI with password encryption set to BCRYPT."
+	@priority = "5"
+	test LoginWithPasswordEncryptionSetToBCRYPT {
+		property custom.properties = "passwords.encryption.algorithm=BCRYPT";
+		property test.name.skip.portal.instance = "Login#LoginWithPasswordEncryptionSetToBCRYPT";
+
+		task ("Login and go to control panel - roles") {
+			User.firstLoginUI();
+
+			ApplicationsMenu.gotoPortlet(
+				category = "Users",
+				panel = "Control Panel",
+				portlet = "Roles");
+		}
+	}
+
+	@description = "Users could login through the UI with password encryption set to BCRYPT to virtual instance."
+	@priority = "5"
+	test LoginWithPasswordEncryptionSetToBCRYPTtoVirtualInstance {
+		property custom.properties = "passwords.encryption.algorithm=BCRYPT";
+		property test.name.skip.portal.instance = "Login#LoginWithPasswordEncryptionSetToBCRYPTtoVirtualInstance";
+
+		task ("Given: Instance admin is login") {
+			User.firstLoginUI();
+		}
+
+		task ("When: Instance admin adds a virtual instance as able.com") {
+			HeadlessPortalInstanceAPI.addPortalInstance(
+				domain = "www.able.com",
+				portalInstanceId = "www.able.com",
+				virtualHost = "www.able.com");
+
+			User.logoutPG();
+
+			User.viewLoggedOutPG();
+		}
+
+		task ("Then: We can to login to the new Virtual Isntance at able.com:8080 successfully") {
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "test@www.able.com");
+
+			ApplicationsMenu.gotoPortlet(
```

Comment:
Prefer to use Role.openRolesAdmin(); instead of using ApplicationsMenu.gotoPortlet